ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/TransactionLogBloomCacher.java

```diff
@@ -202,14 +206,15 @@ private boolean populateLatestSegment(final long eventBlockNumber) {
     return false;
   }
 
-  private void ensurePreviousSegmentsArePresent(final long blockNumber) {
+  public void ensurePreviousSegmentsArePresent(
+      final long blockNumber, final boolean overrideCacheCheck) {
     if (!cachingStatus.isCaching()) {
       scheduler.scheduleFutureTask(
           () -> {
             long currentSegment = (blockNumber / BLOCKS_PER_BLOOM_CACHE) - 1;
             while (currentSegment > 0) {
               try {
-                if (!cachedSegments.getOrDefault(currentSegment, false)) {
+                if (overrideCacheCheck || !cachedSegments.getOrDefault(currentSegment, false)) {
```

Comment:
Connected to my point above about forcing regeneration, why can't we use `cacheLogsBloomForBlockHeader`?

Reply:
Not sure we want to force generation in any case, will verify.